### DISCUSSION TITLE
Slider gets stuck when working its value in controlled mode

### DISCUSSION THREAD

**Discussion by johnny-kc-wu** [chakra-ui/chakra-ui#4464] [Q&A] [answered]
[My Sandbox](https://codesandbox.io/s/slider-in-controlled-mode-xbnpp?file=/components/Panel.js)

So on my next.js project, I try to combine a text input with a slider, and make both of thier value in controlled mode, just like [the example](https://chakra-ui.com/docs/form/number-input#combining-it-with-a-slider) in the document. The problem is that I can update the 'value' or thumb position of the slider by using the buttons or text input field, but the slider in controlled mode is geting stuck when I try to drag the slider bar.

Am I doing wrong on the state management? or any of slider props I misused?

**Comment by payapula** [comment, marked as answer]
@johnny-kc-wu  You are missing `onChange` handler on the slider.

```jsx
  <Slider
          name="slider"
          focusThumbOnChange={false}
          // defaultValue={0}
          value={seek.value}
          max={seek.max}
          min={seek.min}
          step={seek.step}
          onChange={sliderSeekOnChangeEnd} <-- After adding this I can drag the slider
       >
```

Here is my forked [example](https://codesandbox.io/s/slider-in-controlled-mode-forked-omgqh?file=/components/Panel.js)

I removed `onChangeEnd` prop.

BTW, I like your UI  ❤️

**Reply by johnny-kc-wu** [reply]
The [example](https://chakra-ui.com/docs/form/slider#getting-the-final-value-when-dragging-the-slider) from the document uses `onChangeEnd` prop alone. I thought `onChange` can be replaced with `onChangeEnd`, can't it? What if I only want updating the state after sliding is complete?

**Reply by payapula** [reply]
Actually, in the example, that Slider is a uncontrolled component (there is no `value` prop to it) so react would take care of the updates. 

In our case, we are passing `value` to the slider and telling react that we would control it. Now in order for Slider to react, we need `onChange` function to update the `value`. The reason for this is there is another control `Input` that can modify the `value` of the Slider.

Hope it makes sense 😄
